### PR TITLE
Quill Editor - Design update

### LINKED ISSUE #4122: Design System: Quill Editor
## Description

- Thread, Snapshot, and Proposal editor and all states (default, disabled, focus)
  - Markdown on and off
- Comment editor and all states (default, disabled, focus)
  - Markdown on and off
- Reply editor and all states (default, disabled, focus)
  - Markdown on and off

Screen Shot 2023-06-01 at 1 21 29 PM

Figma: https://www.figma.com/file/eIVp33a1oCu0AtcLwSbGjr/%F0%9F%9A%A7-Components-and-Patterns?type=design&node-id=342-4799&t=CMXwC3DWDgOEslre-0

## Acceptance Criteria

- Updates applied to component across the site
- Updates auditable via storybook stories

### PR DESCRIPTION
**Pull request by masvelio**
- Adjusted existing quill editor with [new designs](https://www.figma.com/file/eIVp33a1oCu0AtcLwSbGjr/%F0%9F%9A%A7-Components-and-Patterns?type=design&node-id=342-4799&t=CMXwC3DWDgOEslre-0)
- Adjusted it in all places in the app
  - creating thread
  - editing thread
  - creating comment
  - editing comment
  - replying comment
  - creating new topic
  - editing topic
  - editing profile

## Link to Issue
closes #4122 

image

## Test Plan
- in places listed above check if the editor looks and works as desired

## Outstanding things
I had problems with adding Quill component to the Storybook. I created new ticket which explains the steps I took to try to solve the problem here #4381

### CODE REVIEW

**Comment by zakhap** [comment]
Looking really good, but didn't get a chance to inspect the pixel details. Will have a more thorough review on monday 👍

**Comment by masvelio** [comment]
@meeshlin Just realised that the icon for todo list is missing in figma (or I can't find it). Please provide it

image

**Comment by masvelio** [comment]
@zakhap @meeshlin phosphor icons added, ready to be re-reviewed

**Comment by zakhap** [comment]
I think the new icons look fantastic! Thanks for updating them @masvelio!

My one question is around the expected/desired behavior of the Markdown button, is this correct @meeshlin ? If so, I think this PR is ready to go 🔥 

Screen Shot 2023-07-06 at 9 21 35 AM
Screen Shot 2023-07-06 at 9 21 37 AM

**Comment by meeshlin** [comment]
**Design review**
1). The purpose of having a markdown button is so users know that they're writing in markdown and the non-markdown mode should show the text in their markdown edited format (headers would should up as headers without the # seen in markdown). 
2). In markdown mode, the list buttons (bullet, number, and check) aren't working to automatically add the markdown syntax.

**Important fast follows:** 
1). Hide Placeholder copy — it makes sense to have placeholder text built into the component as placeholders are necessary from an accessibility standpoint and for product/feature function clarity, but we need to hide it for now as we don’t have placeholder copy in our editors
2). Add active state for icon buttons ([Figma dev mode file](https://www.figma.com/file/eIVp33a1oCu0AtcLwSbGjr/%F0%9F%9A%A7-Components-and-Patterns?type=design&node-id=1629%3A12277&mode=dev)) 

@masvelio @zakhap

**Comment by masvelio** [comment]
> 2). In markdown mode, the list buttons (bullet, number, and check) aren't working to automatically add the markdown syntax.

It works but only if you select the text first, which is not ideal. This behaviour exists in the production now (was not introduced in this PR) so I would say that this PR should go in and the bug should be ticketed separately.

**Comment by zakhap** [comment]
> > 2). In markdown mode, the list buttons (bullet, number, and check) aren't working to automatically add the markdown syntax.
> 
> It works but only if you select the text first, which is not ideal. This behaviour exists in the production now (was not introduced in this PR) so I would say that this PR should go in and the bug should be ticketed separately => #4427

I agree with @masvelio here. Let's get this in and then fix the bug subsequently.

File: packages/commonwealth/client/scripts/views/components/component_kit/cw_component_showcase.tsx
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/components/component_kit/cw_component_showcase.tsx:1]: We should adjust this file to use single quotes.

File: packages/commonwealth/client/scripts/views/components/react_quill_editor/react_quill_editor.tsx [hunk 1]
```diff
@@ -204,50 +206,26 @@ const ReactQuillEditor = ({
...
+            toolbarId={toolbarId}
+            isMarkdownEnabled={isMarkdownEnabled}
+            handleToggleMarkdown={handleToggleMarkdown}
+            setIsPreviewVisible={setIsPreviewVisible}
+            isDisabled={isDisabled}
+          />
           <ReactQuill
             ref={editorRef}
-            className={`QuillEditor ${className}`}
+            className={clsx('QuillEditor', className, {
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/components/react_quill_editor/react_quill_editor.tsx:220]: What does clsx do?

**Review by jnaviask — CHANGES_REQUESTED** [review]
Please fix merge conflict.

**Comment by jnaviask** [comment]
Also added #4453 as a known bug present on production. This PR is good to go.

**Pull request merged**